src/pgduckdb_types.cpp

```diff
@@ -1506,10 +1506,10 @@ AppendString(duckdb::Vector &result, Datum value, idx_t offset, bool is_bpchar)
 static void
 AppendJsonb(duckdb::Vector &result, Datum value, idx_t offset) {
 	auto jsonb = DatumGetJsonbP(value);
-	auto jsonb_str = JsonbToCString(NULL, &jsonb->root, VARSIZE(jsonb));
-	duckdb::string_t str(jsonb_str);
+	StringInfo str = makeStringInfo();
+	auto json_str = JsonbToCString(str, &jsonb->root, VARSIZE(jsonb));
```

Comment:
nice change to remove the additional copy. This change does make me realize that we need to wrap AppendJsonb in a PostgresFunctionGuard. And probably we need the same for the list one, since that uses also allocates right?

Reply:
Actually it doesn't seem to save an additional copy. AddString internally will still create a duckdb::string_t out of the `json_str` + `str->len`. But it does save an additional `strlen`, so this still seems like a good change.

Reply:
I believe we saved one copy of the string. The conversion steps are as follows:

1. `JsonBToString` creates the string buffer.
2. `duckdb::string_t str(jsonb_str);` duplicates the string buffer into a C++ string.
3. `AppendString` performs the string -> string copy internally.

With this PR, step 2 is eliminated. For LIST conversion, I wrap the PostgreSQL function with the guard. Do you think we need a separate function, perhaps `AppendList`?